src/InvalidTypeException.php

```diff
@@ -0,0 +1,37 @@
+<?php
+declare(strict_types=1);
+
+namespace Crell\Tukio;
+
+use \ReflectionClass;
+use \ReflectionFunction;
+use \ReflectionException;
+
+class InvalidTypeException extends \RuntimeException
+{
+    protected $message = 'Function does not specify a valid type';
+
+    public function setMessageFromClass($class, string $method)
```

Comment:
These would be better implemented as static function constructors rather than after-the-fact methods.  And should also allow for a parent exception, because you should always retain the exception chain.

So something like (untested):

```php
public static function fromClassCallable($class, string $method, $previous): self
{
   $message = static::$message . "...";
    // ...
    return new static($message..., $previous);
}

// Then
throw InvalidTypeException::fromClass($class, $method, $e);
```

Reply:
Ah, I like that pattern. I'll update and test accordingly.